crates/yttrium/src/chain_abstraction/client.rs

```diff
@@ -396,13 +397,66 @@ impl Client {
         }
     }
 
-    // pub async fn send() {
-    //     // TODO input signed txns
-    //     // TODO send route first, etc. (possibly in parallel)
-    //     // TODO wait_for_success
-    //     // TODO send initial transaction
-    //     // TODO await initial transaction success
-    // }
+    pub async fn execute(
+        &self,
+        ui_fields: UiFields,
+        route_txn_sigs: Vec<PrimitiveSignature>,
+        initial_txn_sig: PrimitiveSignature,
+    ) -> ExecuteDetails {
+        assert_eq!(
+            ui_fields.route.len(),
+            route_txn_sigs.len(),
+            "route_txn_sigs length must match route length"
+        );
+        for (txn, sig) in
+            ui_fields.route.into_iter().zip(route_txn_sigs.into_iter())
+        {
+            let provider = self
+                .provider_pool
+                .get_provider(&txn.transaction.chain_id)
+                .await;
+            let signed = txn.transaction.into_eip1559().into_signed(sig);
+            assert!(provider
+                .send_tx_envelope(TxEnvelope::Eip1559(signed))
+                .await
+                .unwrap()
+                .with_timeout(Some(Duration::from_secs(15)))
+                .get_receipt()
+                .await
+                .unwrap()
+                .status());
+        }
+
+        let _success = self
+            .wait_for_success(
+                ui_fields.route_response.orchestration_id,
+                Duration::from_millis(
+                    ui_fields.route_response.metadata.check_in,
+                ),
+            )
+            .await
+            .unwrap();
+
+        let provider = self
+            .provider_pool
+            .get_provider(&ui_fields.initial.transaction.chain_id)
+            .await;
+        let signed = ui_fields
+            .initial
+            .transaction
+            .into_eip1559()
+            .into_signed(initial_txn_sig);
+        let initial_txn_receipt = provider
+            .send_tx_envelope(TxEnvelope::Eip1559(signed))
+            .await
+            .unwrap()
+            .with_timeout(Some(Duration::from_secs(15)))
+            .get_receipt()
```

Comment:
we should check the `status` field in the receipt to avoid false positive results
see Kotlin: https://github.com/reown-com/reown-kotlin/pull/48/files#diff-8d47957e122c709b23a8a7a640073a656436da862fcf03fde306198618243531R212

check specs: https://docs.metamask.io/services/reference/ethereum/json-rpc-methods/eth_gettransactionreceipt/
`status: Either 1 (success) or 0 (failure)`

Reply:
Oops, yes I missed that for the initial txn, good catch!

I have this for the route txn already tho

Reply:
Fixed in 5dab721bfc1e62f045c4cdd43a16e572bcaa9483